.pre-commit-config.yaml

```diff
@@ -0,0 +1,41 @@
+repos:
+  - repo: https://github.com/PyCQA/flake8
+    rev: 6.0.0
+    hooks:
+      - id: flake8
+        additional_dependencies:
+          - flake8-executable
+          - bandit
+          - flake8-bugbear
+          - flake8-comprehensions
+          - flake8-simplify
+        args: [ "--max-line-length", "120" ]
+        exclude: "tests/to-be-tested/"
+  - repo: https://github.com/pre-commit/mirrors-mypy
+    rev: 'v0.991'
+    hooks:
+      - id: mypy
+        args: ["--disallow-untyped-defs", "--show-error-codes"]
+        exclude: "tests/"
+  - repo: https://github.com/psf/black
```

Comment:
One major reason I left out black mostly aligns with what is said in this article: https://luminousmen.com/post/my-unpopular-opinion-about-black-code-formatter

Code style consistency is more important when a large number of developers are involved to reduce project management overhead. In this small projects, I think we are better off with developers writing things in a manner that is more semantically sensible. flake8 has some basic code style linting and I personally think they should be sufficient.

What do you think?

Reply:
Just to give a quick example of what I meant. Given the following code snippet:

```python
cmd_args = ['-a', 'true',
            '-b', '23',
            '-c', 'a_string']
```

The format above semantically makes sense but `black` would reformat the lines to break after different elements depending on the line length. When a project involves many developers, `black` would make sense because it's just hard to get everyone to be strict on code style -- we would trade some semantically quirky lines formatted by `black` for easier management. But for a small project, this tradeoff won't make sense.

Reply:
Thanks for sharing your line of thought. In the end you can decide how you prefer repository management, linting etc. I'm most interested in merging #66. For the other PRs I'm keen on exchanging ideas, but in the end it is an internal issue and not that relevant for users.  

(I'd like to implement the same functionality here once merged https://github.com/charliermarsh/ruff#flake8-executable-exe)

Reply:
Thanks, feel free to discuss anything you'd like -- I'm also interested in exchanging ideas.

> (I'd like to implement the same functionality here once merged https://github.com/charliermarsh/ruff#flake8-executable-exe)

Not sure what this means?

Reply:
Ruff reimplements flake8 and plugins, amongst other things, in rust. The current EXE001-005 are already there. When EXE006 and EXE007 are there, they can also be included in ruff.